src/v/datalake/coordinator/coordinator.cc

```diff
@@ -10,6 +10,7 @@
 #include "datalake/coordinator/coordinator.h"
 
 #include "base/vlog.h"
+#include "cluster/topic_table.h"
```

Comment:
> Just noting that it'd be great if we could avoid the dependency on cluster if we can. Not a hard requirement (and one that doesn't matter with CMake), but thought I'd call it out since I've been trying to avoid cluster bits in this class

I'm not sure how this will work? Topic table is the source of truth for topic information in the cluster, which is needed for the coordinator. We could reverse the dependency and make cluster call datalake, but this seems more convoluted to me, also topic table is more fundamental IMO (topic_table without the coordinator makes sense, coordinator without topics - not much)